pkg/solana/chainreader/event_read_binding.go

```diff
@@ -0,0 +1,229 @@
+package chainreader
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/gagliardetto/solana-go"
+
+	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logpoller"
+)
+
+type eventReadBinding struct {
+	namespace, genericName string
+	codec                  types.RemoteCodec
+	modifier               commoncodec.Modifier
+	key                    solana.PublicKey
+	remapper               remapHelper
+	indexedSubkeys         map[string]string
```

Comment:
I would lean towards calling this `subKeys` rather than `indexedSubkeys`. The latter might give someone the impression that we support non-indexed subKeys.

Reply:
This comes from the config where the name is `IndexedSubkeys` where I think it is good to imply they are indexed.